invenio_swh/service.py

```diff
@@ -216,9 +215,12 @@ def _parse_status(self, status):
         raise ValueError(f"Invalid status: {status}")
 
     @unit_of_work()
-    def handle_status_update(self, deposit: SWHDeposit, status, uow=None):
+    def update_status(self, deposit: SWHDeposit, status, uow=None):
```

Comment:
comment/shelve: at some point it would be nice to add transactional consistent state changes, i.e. optimistic locking. That means that e.g. when you update a row you do:

```sql
update swh_deposit
set status = 'WAITING'
where 
  id = 123
  and status = 'PENDING';
```

and then check if any rows were updated (i.e. what we already [do for records with the `version_id` field](https://docs.sqlalchemy.org/en/20/orm/versioning.html)).

Reply:
I like this, and will add it afterwards. First I will focus on getting the "must-haves" merged, and then I can return to it.

(I already have some code to improve the whole state management, but I will add it to another PR).

Created an issue:

https://github.com/inveniosoftware/invenio-swh/issues/38